subsys/event_manager/event_manager.c

```diff
@@ -201,6 +217,11 @@ static void event_processor_fn(struct k_work *work)
 			}
 		}
 
+		event_manager_event_postprocess_1(eh);
+		event_manager_event_postprocess_2(eh);
+		event_manager_event_postprocess_3(eh);
+		event_manager_event_postprocess_4(eh);
+
 		event_manager_trace_event_execution(eh, false);
 
```

Comment:
Don't you think this is going into wrong direction? We are adding yet another `__weak`. From my point of view this isn't much different than having event subscribed FINAL. The only difference is that this hook will always be invoked no matter if the event was consumed (keep in mind we already talked about removal of event consumption. This would be exactly the same after that).

I see another solution here.
We could add more dedicated section prefixes (in addition to [this](https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/event_manager/event_manager_priv.h#L31:L35)) and reserve some for such functionalities as we need now (tracer is an example). We would need to agree what should be where (especially when it comes to `power_manager` and `tracer`).
Having this like so would allow us to add CAF tracer module.

The only additional work is to add `SUBSCRIBE_ALL_EVENTS` macro that would be a replacement for those hooks.

Reply:
by "what should be where" i mean which priorities would be reserved for which modules.